OpenRA.Game/Graphics/HardwarePalette.cs

```diff
@@ -60,6 +61,14 @@ public void AddPalette(string name, ImmutablePalette p, bool allowModifiers)
 			int index = palettes.Count;
 			indices.Add(name, index);
 			palettes.Add(name, p);
+
+			if (palettes.Count > buffer.GetLength(0))
+			{
+				var oldBuffer = buffer;
+				buffer = new uint[Exts.NextPowerOf2(palettes.Count), Palette.Size];
```

Comment:
We have a `Texture.SetData(byte[], int, int)`, and it looks like `IPalette.CopyToArray` should already handle copying into a `byte[]` array.  It would break on big-endian systems, but the code already fails at this in a bunch of other places.